### PR TITLE
Renamed `getAbsoluteRestPose` -> `getAbsoluteBindPoseToRef` + related vars.

### PR DESCRIPTION
**Pull request by reimund**
I figured since this private method uses `getBindPose` that should be reflected in the name. Also seems like `ToRef` is the naming convention that should be used.

Also renamed related variables to follow this change.

### CODE REVIEW

**Comment by reimund** [comment]
Damnit, I have my IDE set up to use the same indentation as the file, but it never seems to work.

File: src/Debug/skeletonViewer.ts [hunk 1]
```diff
@@ -560,13 +560,14 @@ export class SkeletonViewer {
         }
     }
 
-    private getAbsoluteRestPose(bone: Nullable<Bone>, matrix: Matrix) {
+    /** function to get the absolute bind pose of a bone by accumulating transformations up the bone hierarchy. */
+    private getAbsoluteBindPoseToRef(bone: Nullable<Bone>, matrix: Matrix) {
```
Comment (Popov72) [inline src/Debug/skeletonViewer.ts:564]: nit: private functions should have a leading underscore in their name (I think we missed this one in the previous PR).

**Comment by Popov72** [comment]
Apart from a nit, LGTM.

File: src/Debug/skeletonViewer.ts [hunk 2]
```diff
@@ -611,18 +612,18 @@ export class SkeletonViewer {
...
                 let anchorPoint = new Vector3();
 
-                boneAbsoluteRestTransform.decompose(undefined, undefined, anchorPoint);
+                boneAbsoluteBindPoseTransform.decompose(undefined, undefined, anchorPoint);
 
                 bone.children.forEach((bc, i) => {
-                    let childAbsoluteRestTransform : Matrix = new Matrix();
-                    bc.getRestPose().multiplyToRef(boneAbsoluteRestTransform, childAbsoluteRestTransform);
+                    let childAbsoluteRestPoseTransform : Matrix = new Matrix();
+                    bc.getRestPose().multiplyToRef(boneAbsoluteBindPoseTransform, childAbsoluteRestPoseTransform);
```
Comment (reimund) [inline src/Debug/skeletonViewer.ts:624]: Is it correct to use `getRestPose` here? In the helper function we use `getBindPose`. Is there a reason for not using the the same when figuring out the child bone position?

Comment (deltakosh) [reply src/Debug/skeletonViewer.ts:624]: @Pryme8  / @Popov72  / @CedricGuillemet

**Comment by Pryme8** [comment]
I am not sure about this part.  I think @popov72

File: src/Debug/skeletonViewer.ts (hunk 1 above)
Comment (Popov72) [reply src/Debug/skeletonViewer.ts:564]: Yes, as it is computing an absolute bind pose matrix it should start by using the local bind matrix, not the rest pose.

File: src/Debug/skeletonViewer.ts (hunk 2 above)
Comment (Popov72) [reply src/Debug/skeletonViewer.ts:624]: Yes, as it is computing an absolute bind pose matrix it should start by using the local bind matrix, not the rest pose.

Comment (reimund) [reply src/Debug/skeletonViewer.ts:624]: Are you saying that it should be changed to `getBindPose`?

Comment (Popov72) [reply src/Debug/skeletonViewer.ts:624]: yes

**Comment by Pryme8** [comment]
This is not breaking any backwards compatibility is it?

**Comment by Popov72** [comment]
I guess some testing should be performed to check it is ok.

Note that there's no really backward compatibility problem as if I'm not mistaken the part of the skeleton viewer which is using this code is new in 4.2.

**Comment by Pryme8** [comment]
> I guess some testing should be performed to check it is ok.
> 
> Note that there's no really backward compatibility problem as if I'm not mistaken the part of the skeleton viewer which is using this code is new in 4.2.

I think that is for the lines view isn't it?   I'm pretty sure I never used it in the new stuff.

**Comment by deltakosh** [comment]
Are we good to merge then?

**Comment by reimund** [comment]
> Are we good to merge then?

I think so!

**Pull request merged**